File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    const stack=[]
    const openedBrackets=['(','[','{']
    for(let str of s){
        if(stack.length===0){
            if(openedBrackets.includes(str)){
                stack.push(str)
            }else{
                return false
            }
        }else{
            if(openedBrackets.includes(str)){
                stack.push(str)
            }else{
                if(str===')'){
                    if(stack[stack.length-1]==='('){
                        stack.pop()
                    }else{
                        return false
                    }
                }else if(str===']'){
                    if(stack[stack.length-1]==='['){
                        stack.pop()
                    }else{
                        return false
                    }
                }else if(str==='}'){
                    if(stack[stack.length-1]==='{'){
                        stack.pop()
                    }else{
                        return false
                    }
                }
            }
        }
    }
    if(stack.length>0) return false
    return true
};

File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        vector<char> stack;
        for(int i=0;i<(int)s.size();i++){
            std::cout << s[i] << "\n";
            if(s[i]=='('||s[i]=='{'||s[i]=='['){
                stack.push_back(s[i]);
            }else{
                if(stack.size()==0) return false;
                if(s[i]==')'){
                    if(stack.back()=='('){
                        stack.pop_back();
                    }else return false;
                }else if(s[i]=='}'){
                    if(stack.back()=='{'){
                        stack.pop_back();
                    }else return false;
                }else if(s[i]==']'){
                    if(stack.back()=='['){
                        stack.pop_back();
                    }else return false;
                }
            }
        }
        if(stack.size()!=0) return false;
        return true;
    }
};